lib/src/parser/recursive.dart

```diff
@@ -425,136 +443,23 @@ class RecursiveAstParser {
         properties: properties,
         references: references,
         bananas: bananas,
-        stars: stars,
+        stars: (uniqueStarAst == null ? [] : [uniqueStarAst]),
```

Comment:
Sounds like a good idea. I'll probably incorporate this in a separate PR with all the necessary checks needed for this.